delivery_carrier_label_paazl/models/stock_picking.py

```diff
@@ -0,0 +1,224 @@
+        wsdl = self.env["ir.config_parameter"].get_param(
+            "delivery_carrier_label_paazl.wsdl_%s"
+            % (production and "production" or "test")
+        )
+        return zeep.Client(wsdl)
+
+    def _paazl_raise_error(self, api_result):
+        """If paazl's API returned an error, raise it"""
+        if api_result.error:
+            raise UserError(
+                _("Sending %s to paazl: %s %s")
+                % (self.name, api_result.error.code, api_result.error.message,)
+            )
+
+    def _paazl_order_data(self):
+        """Return a dict that can be passed as parameters to the `order`
+        endpoint of the paazl soap api"""
+        self.ensure_one()
+        return dict(
+            **self._paazl_auth_order_id(),
+            products=dict(
+                product=[
+                    self._paazl_quant_to_order_data(quant)
+                    for package in self.package_ids
+                    for quant in package.quant_ids
+                ]
+            ),
+        )
+
+    def _paazl_quant_to_order_data(self, quant):
+        """Return a dict describing a quant for the `order` endpoint"""
+        result = {
+            "quantity": int(quant.quantity),
+            "weight": quant.product_id.weight,
+            "volume": quant.product_id.volume,
+            "code": quant.product_id.default_code,
+            "description": quant.product_id.description_pickingout or "/",
+            "hsTariffCode": quant.product_id.hs_code,
+        }
+        # support product_dimension if installed
+        for name in ["length", "height", "width"]:
+            field_name = "product_%s" % name
+            if field_name not in quant.product_id._fields:
+                continue
+            result[name] = quant.product_id[field_name]
+            uom = quant.product_id.dimensional_uom_id
+            if uom:
+                result[name] = uom._compute_quantity(
+                    result[name], self.env.ref("uom.product_uom_cm"),
+                )
+            result[name] = int(result[name])
+        return result
+
+    def _paazl_commit_order_data(self):
+        """Return a dict that can be passed as parameters to the `commitOrder`
+        endpoint of the paazl soap api"""
+        self.ensure_one()
+        shipping_address = self._paazl_partner_to_order_data(self.partner_id)
+        shipping_address["customerName"] = shipping_address.pop("addresseeLine")
+        if self.partner_id.parent_id:
+            shipping_address["companyName"] = self.partner_id.parent_id.name
+        return dict(
+            **self._paazl_auth_order_id(),
+            pendingOrderReference=self._paazl_reference(),
+            customerEmail=self.partner_id.email,
+            customerPhoneNumber=self.partner_id.phone,
+            shippingMethod=dict(
+                type="delivery",
+                identifier=0,
+                # TODO should we validate that there's exacly one option set?
```

Comment:
Yes. I'm actually amazed that options cannot be deactivated. If you agree that that would be useful, is that something you can pick up in a separate PR on base_delivery_carrier_label?

Reply:
I'd like to overhaul the way options work entirely, that's one of the things I'd like to discuss after the review

Reply:
@StefanRijnhart I added a check for options in https://github.com/VanMoof/delivery-carrier/pull/1/commits/53a64f78b5fe45de2180372867893dc0ef92d63b - if you agree with this, I'll squash and PR to OCA. This can't be a constraint because the carrier is set on the picking before we have a chance to set options.

The active flag is added in https://github.com/OCA/delivery-carrier/pull/288

Reply:
Yes, agreed.